kyaml/fn/runtime/runtimeutil/functiontypes.go

```diff
@@ -200,50 +201,51 @@ func (s *StorageMount) String() string {
 //
 // The FunctionSpec is read from the resource metadata.annotation
 // "config.kubernetes.io/function"
-func GetFunctionSpec(n *yaml.RNode) *FunctionSpec {
+func GetFunctionSpec(n *yaml.RNode) (*FunctionSpec, error) {
 	meta, err := n.GetMeta()
 	if err != nil {
-		return nil
+		return nil, nil
 	}
-	if fn := getFunctionSpecFromAnnotation(n, meta); fn != nil {
-		return fn
+	if fn, err := getFunctionSpecFromAnnotation(n, meta); err != nil {
+		return nil, err
+	} else if fn != nil {
```

Comment:
I'm surprised the linter didn't complain about this.

We prefer this style: 

```
if fn, err := getFunctionSpecFromAnnotation(n, meta); err != nil {
		return nil, err
} 
if fn != nil {
...
```

the `else if` is not needed because the previous statement is returning.